sdk/core/System.ClientModel/gen/ModelReaderWriterContextGenerator.cs

```diff
@@ -458,6 +477,9 @@ private static bool IsModelReaderWriterContext(ClassDeclarationSyntax classDecla
                 return persistableSymbol;
             }
 
+            if (baseTypeSymbol.TypeKind != TypeKind.Interface)
```

Comment:
If this check is true, would it also have failed the previous check? If so, would it be more efficient to put this before the `if (IsIPersistableInterface(baseTypeSymbol))` check?

Reply:
We want to scan hierarchy only for interfaces.  If you have an interface not a base class.

For example
```c#
public class Foo : Bar {}

public class Bar : IPersistableModel<Bar> {}
```

We don't want to make a builder for Foo since itself doesn't implement IPersistableModel

That being said if you have
```c#
public class Foo : IJsonModel<Foo> {}
```

Needs to be caught, it will wail the IsIPersistableInterface because it is not, but IJsonModel inherits from IPersistableModel and therefore should still be treated as IPersistableModel.